sdk/identity/azure-identity/src/main/java/com/azure/identity/credential/AadCredentialBuilderBase.java

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.identity.credential;
+
+/**
+ * The base class for credential builders that allow specifying a client id and tenant ID for an Azure Active Directory.
+ * @param <T> the type of the credential builder
+ */
+class AadCredentialBuilderBase<T extends AadCredentialBuilderBase<T>> extends CredentialBuilderBase<T> {
```

Comment:
It can sometimes be weird to have public API that extends a base class that itself is not public API. I would be inclined to instead make this class public and abstract. Thoughts?